packages/devtools_server/lib/src/chrome.dart

```diff
@@ -55,35 +54,35 @@ class Chrome {
 
   /// Connects to an instance of Chrome with an open debug port.
   static Future<Chrome> fromExisting(int port) async =>
-      _connect(Chrome._(port, ChromeConnection('localhost', port)));
+      _connect(Chrome._(ChromeConnection('localhost', port), debugPort: port));
 
   static Future<Chrome> get connectedInstance => _currentCompleter.future;
 
-  /// Starts Chrome with the remote debug port enabled.
+  /// Starts Chrome with the given arguments and a specific port.
   ///
   /// Each url in [urls] will be loaded in a separate tab.
-  static Future<Chrome> start(List<String> urls, {int port}) async {
-    var dataDir = Directory.systemTemp.createTempSync();
+  static Future<Chrome> startWithPort(
+    List<String> urls, {
+    List<String> args,
+    int port,
+  }) async {
     port = port == null || port == 0 ? await findUnusedPort() : port;
-    var args = [
-      // Using a tmp directory ensures that a new instance of chrome launches
-      // allowing for the remote debug port to be enabled.
-      '--user-data-dir=${dataDir.path}',
-      '--remote-debugging-port=$port',
-      // When the DevTools has focus we don't want to slow down the application.
-      '--disable-background-timer-throttling',
-      // Since we are using a temp profile, disable features that slow the
-      // Chrome launch.
-      '--disable-extensions',
-      '--disable-popup-blocking',
-      '--bwsi',
-      '--no-first-run',
-      '--no-default-browser-check',
-      '--disable-default-apps',
-      '--disable-translate',
-    ]..addAll(urls);
-
-    var process = await Process.start(_executable, args);
+    return start(urls, args: args, port: port);
+  }
+
+  /// Starts Chrome with the given arguments.
+  ///
+  /// Each url in [urls] will be loaded in a separate tab.
+  static Future<Chrome> start(
+    List<String> urls, {
+    List<String> args,
+    int port,
+  }) async {
+    final dataDir = Directory.systemTemp.createTempSync();
+    args ??= [];
```

Comment:
use Dart default arguments instead.

Reply:
oddly enough, changing this to `List<String> args = const [],` breaks everything 🤔

Reply:
you can't add elements to a const list. Create a new list with the args to pass to Process.start. 
In general, it is bad practice to modify arguments passed in to a method when you don't need to so that way of writing the code is cleaner in general.